### ISSUE TITLE
Feature request: pandas.Series.query()

### ISSUE THREAD

**Issue by adamjstewart** [pandas-dev/pandas#22347]
I would like to request a `pandas.Series.query()` method that works identically to [`pandas.DataFrame.query()`](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.query.html).

I have a large multi-indexed series that I would like to split into training and testing data for an ML research project. A minimal working example would be:
```python
import pandas as pd

years = range(2002, 2018)
fields = range(1, 5)

index = pd.MultiIndex.from_product(
    [years, fields], names=['year', 'field'])

series = pd.Series(index=index)
```
What I would like to be able to do is split this series into 2010 data and _not_ 2010 data. Accessing 2010 data is very easy:
```python
test_data = series[2010]
```
Accessing _not_ 2010 data is very hard. This is the shortest method I've found so far:
```python
train_data = series.to_frame().query('year != 2010')[0]
```
Since `pandas.Series` doesn't support `query()`, I have to convert it to a DataFrame and then back into a Series. Is there any reason why `pandas.Series` doesn't support `query` directly?

**Comment by gfyoung** [comment]
Good question!  Not entirely sure myself.

cc @jreback

**Comment by adamjstewart** [comment]
The more I use pandas, the more I find Series lacking the same features as DataFrame. Should I just always use a single column DataFrame instead of a Series?

**Comment by gfyoung** [comment]
You could as a workaround, but that doesn't necessarily mean we get a pass on not adding more features to `Series`. 🙂

**Comment by WillAyd** [comment]
Perhaps a better solution to your problem would be to use a mask based off of the index values, so:

```python
mask = index.get_level_values('year') == 2010
train = series[mask]
test = series[~mask]
```

**Comment by WillAyd** [comment]
For the sake of explicitness I am -1 on adding `query` to `Series` as I don't see it adding much value over the masking solution provided above.

**Comment by gfyoung** [comment]
@WillAyd : I'm confused by your objection here.  How is `.query` not explicit?

**Comment by adamjstewart** [comment]
Thanks @WillAyd, that syntax works identically for both Series and DataFrame, which is exactly what I was looking for.

At the same time, I think supporting different methods for Series and DataFrame makes it very confusing for new users (like myself). When I searched for how to do this with a MultiIndex, the `query` method was the first answer I found. It worked great for splitting my data (DataFrame) but didn't work for splitting my labels (Series). Are you also proposing we drop `query` from DataFrames because the same thing _could_ be done through other methods?

**Comment by WillAyd** [comment]
`.query('year != 2010)` is explicit but also duplicative of functionality that already exists. Coding and maintaining that imo is more of a burden than the incremental value it provides, especially since `Series` objects typically are unnamed.

I am not proposing to drop `query` from DataFrame as it is more useful there were your columns will typically be named. If anything the talk about adding `.query` here to a Series is a red herring as I think the biggest confusion is how to leverage a single level of a MultiIndex in a masking operation.

**Comment by adamjstewart** [comment]
I agree that the masking solution works just fine, but I do still think that having Series and DataFrame support different methods is quite non-intuitive. Especially coming from numpy, where there is no distinction between 1D and 2D arrays, they all support identical operations.

**Comment by adamjstewart** [comment]
Also see https://stackoverflow.com/questions/40171498 for someone else who had the same problem as I did.

**Comment by zachmoshe** [comment]
Just to add another use-case: 

I'd like to easily get only values that appear more than X times in a column:
```
my_series = pd.Series(...)  # or df['col']
my_series.value_counts().query('self > 50')  # how should we refer the value here? series's name? what if it's unnamed? I used 'self' just as a placeholder
```

I obviously want to avoid creating a temp variable for the value counts, and don't like (although that's what I do today) the `to_frame().query(..)` hack.

**Comment by yaolan4** [comment]
take

**Comment by yaolan4** [comment]
take

**Comment by yaolan4** [comment]
take

**Comment by Alalalalaki** [comment]
Any progress on this?

**Comment by jreback** [comment]
@Alalalalaki community contributions are how things get done

**Comment by Alalalalaki** [comment]
@jreback I read the code and find the code of DataFrame's `eval` and `query` functions can be directly applied to Series as long as the series has a name. I am not sure about the 'self' idea now, which is somehow unnatural. Do you think simply copying the code from frame to series is a good idea?

**Comment by jreback** [comment]
don't copy but reuse / share code
